charm/timescaledb/src/charm.py

```diff
@@ -32,18 +32,36 @@ def on_install(self, event):
                 event.framework.model.unit.status = WaitingStatus('waiting for postgresql to be installed')
                 event.defer()
                 return
-            check_call(['add-apt-repository', '-y', 'ppa:timescale/timescaledb-ppa'])
-            check_call(['apt-get', 'update', '-qq'])
-            if os.path.exists('/var/lib/postgresql/10'):
-                check_call(['apt-get', 'install', '-y', 'timescaledb-postgresql-10', 'postgresql-server-dev-10'])
-            elif os.path.exists('/var/lib/postgresql/11'):
-                check_call(['apt-get', 'install', '-y', 'timescaledb-postgresql-11', 'postgresql-server-dev-11'])
+            
+            # install dependencies
+            check_call(['sudo', 'apt-get', 'install', '-y', 'apt-transport-https', 'lsb-release', 'wget'])
+
+            # add apt repo to sources
+            release = check_output(['lsb_release', '-c', '-s']).decode("utf-8").rstrip('\n')
+            ps = Popen(['echo', 'deb https://packagecloud.io/timescale/timescaledb/ubuntu/ {} main'.format(release)], stdout=PIPE)
+            check_call(['sudo', 'tee', '/etc/apt/sources.list.d/timescaledb.list'], stdin=ps.stdout)
+            ps.wait()
+            
+            # add apt key
+            ps = Popen(['wget', '--quiet', '-O', '-', 'https://packagecloud.io/timescale/timescaledb/gpgkey'], stdout=PIPE)
+            check_call(['sudo', 'apt-key', 'add', '-'], stdin=ps.stdout)
+            ps.wait()
+            
+            # install TimescaleDB
+            check_call(['sudo','apt-get', 'update', '-qq'])
+            
+            if os.path.exists('/var/lib/postgresql/12'):
+                pgver = 12
+            elif os.path.exists('/var/lib/postgresql/14'):
+                pgver = 14            
             else:
                 event.framework.model.unit.status = BlockedStatus('failed to find a compatible version of postgresql (10, 11)')
```

Comment:
Versions in the string does not match the actual values anymore. I think it would be ideal to dynamically get the version from a list of compatible PostgreSQL, TimescaleDB version mapping but changing the string is good enough.